### PR TITLE
Configuration UI improvements

### LINKED ISSUE #637: Improve current Configuration UI
This includes vastly improving the UI of configuring stuff in the Monkey.
This does NOT include added a wizard/scenarios feature, only improving the current layout content.

- [ ] Document each config value with examples in the #602.
- [x] #595
- [x] #526
- [x] #566 
- [x] Add validation to fields like network segments (possibly convert `Form` to `Formik`)

### PR DESCRIPTION
**Pull request by VakarisZ**
# What is this? 
Fixes #637

## Checklist
- Ask Shreya to add descriptions to PBA's she added.
- Synch links with new documentation framework.

- Instead of "Basic - ..." we should highlight essential tabs with green colour (**DONE**)
- Create an advanced selector, that allows to select/deselect on click. On click 
information about clicked property is shown in selectors info pane. Advanced selector
also has space for custom buttons. (**DONE**)

## Basic - Exploits

- Should be renamed to Exploits (**DONE**)

### General
- Renamed to "Exploiters" (**DONE**)
- Exploiter selector moved here from "Exploits" tab. (**DONE**)
- "Exploit network machines" is removed (you can deselect all exploiters) (**DONE**)

## Basic - Network

- Should be renamed to Network (**DONE**)

### General 

- Should be renamed to "Scope" (**DONE**)

### Behaviour

- "Self delete on cleanup" - moved to Internal/Monkey (on by default). (**DONE**)
- "Serialize config" - moved to Internal/Monkey (off by default). (**DONE**)
- "Use file logging" - moved to Internal/Monkey (on by default). (**DONE**)
- Renamed to "Post breach" (**DONE**)
 
### General

- "Is monkey alive" - hidden, moved to internal->monkey. (**DONE**)
- "Post breach actions" - advanced selector with info pane that displays 
what each post breach action does. (**DONE**)
- PBA selector moved to "Post breach" (**DONE**)

### Lifecycle

- Renamed to "Persistent scanning"
- "Max victims to exploit" - Moved to internal, increased default to 100. (**DONE**)
- "Max victims to find" - Moved to internal. (**DONE**)

### System info

- "System info collectors" refactored into advanced selector component. (**DONE**)
- "Collect system info" - removed. Same can be achieved by disabling all collectors. (**DONE**)
- "Should use mimikatz" - refactored into Mimikatz system info collector. (**DONE**)
- "Harvest Azure credentials" - refactored into system info collector. (**DONE**)

### Servers

- Whole "Servers" section moved to Internal. "Command servers" uses all local interfaces
which is a reasonable default. "Current server" is also chosen automatically/doesn't matter. (**DONE**)

### Internet services

- Section moved to "Internal -> Monkey" (**DONE**)

## Network

- Whole tab moved to internal (**DONE**)

## Exploits

- Tab moved to be after "Network" (**DONE**)
- Everything in this tab except "Exploits" are moved to "Internal". (**DONE**)

#### Exploits

- "Exploits" made into advanced selector. Info pane displays information about exploiters, there is a button 
that de-selects all exploiters. (**DONE**)

## Internal 

- Refactored into sub-tabs or dropdowns for each section. (**DONE**)

## Proof that it works
image

### CODE REVIEW

File: monkey/monkey_island/cc/services/config_schema/basic.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                    "type": "array",
+                    "uniqueItems": True,
+                    "items": {
+                        "type": "string"
+                    },
+                    "default": [
+                        "Password1!",
+                        "1234",
+                        "password",
+                        "12345678"
```
Comment (ShayNehmad) [inline monkey/monkey_island/cc/services/config_schema/basic.py:64]:
```suggestion
                        "Password1!",
                        "password",
                        "root",
                        "12345678",
                        "qwerty",
                        "iloveyou"
```

Probably time to update the default passwords list? Just an idea

File: monkey/monkey_island/cc/ui/src/components/configuration-components/ValidationFormats.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  'ip-range': buildIpRangeRegex(),
+  'ip': buildIpRegex(),
+};
+
+function buildIpRangeRegex(){
+  return new RegExp([
+    '^'+ipRegex+'$|', // Single IP
+    '^'+ipRegex+'-'+ipRegex+'$|', // IP range IP-IP
+    '^'+ipRegex+'/'+cidrNotationRegex+'$|', // IP range with cidr notation: IP/cidr
+    hostnameRegex, // IP range with cidr notation: IP/cidr
```
Comment (ShayNehmad) [inline monkey/monkey_island/cc/ui/src/components/configuration-components/ValidationFormats.js:15]: copy-pasta documentation

File: monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js [hunk 3]
```diff
@@ -366,110 +324,65 @@ class ConfigurePageComponent extends AuthComponent {
...
+
   renderBasicNetworkWarning = () => {
     if (this.state.selectedSection === 'basic_network') {
       return (<div className='alert alert-info'>
         <FontAwesomeIcon icon={faInfoCircle} style={{'marginRight': '5px'}}/>
-        The Monkey scans its subnet if 'Local network scan' is ticked. Additionally the monkey scans machines
-        according to its range class.
+        The Monkey scans its subnet if "Local network scan" is ticked. Additionally the monkey scans machines
+        according to "Scan target list".
       </div>)
```
Comment (ShayNehmad) [inline monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js:386]: We should move this warning into the "scope" section of the configuration page. It's out of place at the top of the page.

File: monkey/monkey_island/cc/ui/src/components/configuration-components/ValidationFormats.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const ipRegex = '((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)'
```
Comment (ShayNehmad) [inline monkey/monkey_island/cc/ui/src/components/configuration-components/ValidationFormats.js:1]:
```
monkey\monkey\monkey_island\cc\ui\src\components\configuration-components\ValidationFormats.js
  1:57  warning  Unnecessary escape character: \.  no-useless-escape
```

File: monkey/monkey_island/cc/services/config_schema/basic.py (hunk 1 above)
Comment (VakarisZ) [reply monkey/monkey_island/cc/services/config_schema/basic.py:64]: Good idea, updated a bit. Shame that we have limited brute forcing capabilities due to performance and noise generation

**Pull request merged**